### PR TITLE
Feature/steevejoseph/quality of life fixes

### LINKED ISSUE #2: Make pressing Enter do the same thing as pressing the search button
Currently the user must click the button to perform the search, would be better to have this done on enter press. We tried it before, but the enter event fired multiple times, jamming the responses into each other

### PR DESCRIPTION
**Pull request by steevejoseph**
#2

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -27,13 +30,33 @@ func NewApp() *App {
 // so we can call the runtime methods
 func (a *App) startup(ctx context.Context) {
 	a.ctx = ctx
+	fmt.Println("Backend debugging is working!")
```
Comment (lonesume) [inline app.go]: pls remove comment.

File: app.go [hunk 2]
```diff
@@ -27,13 +30,33 @@ func NewApp() *App {
...
+	if err != nil {
+		return ""
+	}
+
+	var name string
+
+	if curUser.Name != "" {
+		name = curUser.Name
+	} else {
+		split := strings.Split(curUser.Username, "/")
```
Comment (lonesume) [inline app.go]: Can you clarify what this is doing specifically and why the if check was needed? It's not clear why `curUser.Name` would be empty.

File: frontend/src/App.tsx [hunk 3]
```diff
@@ -1,46 +1,112 @@
...
+    };
+
+    typeChar();
+  };
+
+  // Handle keyboard events
+  const handleInputEvent = useCallback(
+    async (event: KeyboardEvent) => {
+      const key = event.key;
+      console.log(event);
```
Comment (lonesume) [inline frontend/src/App.tsx]: Is this log needed?

File: frontend/src/App.tsx [hunk 4]
```diff
@@ -1,46 +1,112 @@
...
+
+      switch (key) {
+        case "Enter":
+          if (isGptResponseLoading || isTyping) {
+            return;
+          }
+          await getGPTResponse();
+          return;
+        case "Escape":
+          console.log("closing");
```
Comment (lonesume) [inline frontend/src/App.tsx:72]: Hey, noticed this logged out closing. Did you mean to add that feature to this PR as well?

File: frontend/src/App.tsx [hunk 5]
```diff
@@ -1,46 +1,112 @@
...
+          return;
+      }
+    },
+    [query, isGptResponseLoading, isTyping, getGPTResponse]
+  );
+
+  // Fetch username on component mount
+  useEffect(() => {
+    async function fetchUserName() {
```
Comment (lonesume) [inline frontend/src/App.tsx]: Can we be consistent on usage of `function` vs `const` syntax? Or maybe we can determine which is better to use in what case.

File: frontend/src/App.tsx
Comment (lonesume) [inline frontend/src/App.tsx:1]: We use a lot async await syntax here, can we add try/catch logic to handle errors?

**Review by lonesume — CHANGES_REQUESTED** [review]
We're off to a good start, let's get those adddressed

File: frontend/src/App.tsx (hunk 4 above)
Comment (steevejoseph) [reply frontend/src/App.tsx:72]: Yeah, actually I split this into a separate issue #4

File: frontend/src/App.tsx (hunk 5 above)
Comment (steevejoseph) [reply frontend/src/App.tsx]: True, `const` supremacy

**Review by lonesume — APPROVED** [review]
LGTM. Good soup

**Pull request merged**